src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -507,7 +507,12 @@ public PublishPackageViewModel( DynamoViewModel dynamoViewModel ) : this()
 
         private void ThisPropertyChanged(object sender, PropertyChangedEventArgs e)
         {
-            if (e.PropertyName == "PackageContents") CanSubmit();
+            if (e.PropertyName == "PackageContents")
+            {
+                CanSubmit();
+                BeginInvoke(
```

Comment:
I agree with @ikeough.  It looks like you're using this method defensively.  I cannot conceive of a situation that this will be called from anything but the dispatcher thread to begin with.  This will also remove the nasty `DoEvents()` call in your test.